docs/src/types/jump_types.md

```diff
@@ -23,26 +23,37 @@ Diffusion, and is denoted by
-change the size of `u`, require exact handling which is also known as time-adaptive
-jumping and can only be specified as a `ConstantRateJump` or a `VariableRateJump`.
-
-We denote a jump as variable rate if its rate function is dependent on values which
-may change between constant rate jumps. For example, if there are multiple jumps
-whose rates only change when one of them occur, than that set of jumps is a constant
-rate jump. If the jump's rate depends on the differential equation, time, or
-by some value which changes outside of some constant rate jump, then it is denoted
-as variable.
-
-`RegularJump`s are optimized for regular jumping algorithms like tau-leaping
-and hybrid algorithms. `ConstantRateJump`s are optimized for SSA algorithms.
-`ConstantRateJump`s and `VariableRateJump`s can be added to standard DiffEq
-algorithms since they are simply callbacks, while `RegularJump`s require special
-algorithms.
+## Types of Jumps: Regular, Variable, Constant Rate and Mass Action
+
+A `RegularJump` is a set of jumps that do not make structural changes to the
+underlying equation. These kinds of jumps only change values of the dependent
+variable (`u`) and thus can be treated in an inexact manner. Other jumps, such
+as those which change the size of `u`, require exact handling which is also
+known as time-adaptive jumping. These can only be specified as a
+`ConstantRateJump`, `MassActionJump`, or a `VariableRateJump`.
+
+We denote a jump as variable rate if its rate function is dependent on values
+which may change between constant rate jumps. For example, if there are multiple
+jumps whose rates only change when one of them occur, than that set of jumps is
+a constant rate jump. If a jump's rate depends on the differential equation,
+time, or by some value which changes outside of any constant rate jump, then it
+is denoted as variable.
+
+A `MassActionJump` is a specialized representation for a collection of constant rate
+jumps that can each be interpreted as a standard mass action reaction. For systems
+comprised of many mass action reactions, using the `MassActionJump` type
+will offer improved performance, and can also result in fewer memory allocations. Note,
+only one `MassActionJump` should be defined per `JumpProblem`; it is then responsible for
+handling all mass action reaction type jumps. For systems with both mass action jumps and non-mass
+action jumps, one can create one `MassActionJump` to handle the mass action jumps,
+and create a number of `ConstantRateJumps` to handle the non-mass action jumps.
+
+`RegularJump`s are optimized for regular jumping algorithms like tau-leaping and
+hybrid algorithms. `ConstantRateJump`s and `MassActionJump`s are optimized for
+SSA algorithms. `ConstantRateJump`s and `VariableRateJump`s can
+be added to standard DiffEq algorithms since they are simply callbacks, while
+`RegularJump`s require special algorithms. `MassActionJump`s are currently specialized
```

Comment:
`MassActionJump`s should mix fine with diffeqs as long as they satisfy the constant rate assumption.

Reply:
In theory yes, in practice they need a slight tweak. The type signature of `evalrxrate` and `executrerx!` needs to be modified. Right now they force the species population vector to have the same type as the species index and species stoichiometry coefficient. This is an issue when used with ODEs.

Reply:
Got it. Can that get an issue?

Reply:
Yes, the only reason I didn't put one in is that I was going to fix it myself today.

Reply:
Awesome! :)

Reply:
I put one in anyways, but will still try to fix it today...